recipes-bsp/uefi/tegra-uefi-capsules_35.5.0.bb

```diff
@@ -11,45 +11,88 @@ TEGRA_UEFI_CAPSULE_SIGNING_EXTRA_DEPS ??= ""
 
 COMPATIBLE_MACHINE = "(tegra)"
 
-DEPENDS += "tegra-bup-payload"
+TEGRA_SIGNING_EXTRA_DEPS ??= ""
 
 GUID:tegra194 ?= "be3f5d68-7654-4ed2-838c-2a2faf901a78"
 GUID:tegra234 ?= "bf0d4599-20d4-414e-b2c5-3595b1cda402"
 
 do_compile() {
+    # Generate BUP images
+    PATH="${STAGING_BINDIR_NATIVE}/${FLASHTOOLS_DIR}:${PATH}"
+    export tosimgfilename=${TOSIMGFILENAME}
+    rm -rf ${B}/bup-payload
+    mkdir ${B}/bup-payload
+    oldwd="$PWD"
+    cd ${B}/bup-payload
+    # BUP generator really wants to use 'boot.img' for the LNX
+    # partition contents
+    tegraflash_populate_package ${IMAGE_TEGRAFLASH_KERNEL} boot.img ${@tegra_bootcontrol_overlay_list(d, bup=True)}
+    mv generate_bup_payload.sh doflash.sh
+    tegraflash_create_flash_config flash.xml.in boot.img ${STAGING_DATADIR}/tegraflash/bupgen-internal-flash.xml
+    . ./flashvars
+    tegraflash_custom_sign_bup
+    for bup in ${B}/bup-payload/${BUP_PAYLOAD_DIR}/*; do
+	    [ -e $bup ] || continue
+	    BUP_generator.py --contents --check $bup
+    done
+    mv ${B}/bup-payload/${BUP_PAYLOAD_DIR}/* .
+    cd "$oldwd"
+
+    # Create symlinks BUP payloads with a naming expected by sign_uefi_capsules
+    for f in ${B}/bup-payload/*_only_payload; do
+        [ -e $f ] || continue
+        sfx=$(basename $f _payload)
+
+        ln -sf $f ${B}/${BUPFILENAME}.$sfx.bup-payload
+    done
+
+    # Generate UEFI capsules
     sign_uefi_capsules
+
+    # Check if capsules were generated successfully
+    if [ ! -e ${B}/tegra-bl.cap ]; then
+        bberror "${B}/tegra-bl.cap wasn't generated"
+    fi
+    if [ ! -e ${B}/tegra-kernel.cap ]; then
+        bberror "${B}/tegra-kernel.cap wasn't generated"
+    fi
```

Comment:
Almost... at least for the kernel one, the test should be something like `[ -e ${B}/${BUPFILENAME}.kernel.bup_payload -a ! -e ${B}/tegra-kernel.cap ]`.   There are cases where no kernel BUP will get created, and that's OK.  (No one should be using capsules to update their kernels anyway.)

Reply:
fixed, thanks @madisongh